internal/util/importutilv2/parquet/util.go

```diff
@@ -146,11 +151,49 @@ func isArrowDataTypeConvertible(src arrow.DataType, dst arrow.DataType, field *s
 	case arrow.NULL:
 		// if nullable==true or has set default_value, can use null type
 		return field.GetNullable() || field.GetDefaultValue() != nil
+	case arrow.STRUCT:
+		if field.GetDataType() == schemapb.DataType_SparseFloatVector {
+			valid, _ := IsValidSparseVectorSchema(src)
+			return valid
+		}
+		return false
 	default:
 		return false
 	}
 }
 
+// This method returns two booleans
+// The first boolean value means the arrowType is a valid sparse vector schema
+// The second boolean value: true means the sparse vector is stored as JSON-format string,
+// false means the sparse vector is stored as parquet struct
+func IsValidSparseVectorSchema(arrowType arrow.DataType) (bool, bool) {
+	arrowID := arrowType.ID()
+	if arrowID == arrow.STRUCT {
+		arrType := arrowType.(*arrow.StructType)
+		indicesType, ok1 := arrType.FieldByName(sparseVectorIndice)
```

Comment:
What if the struct doesn't contain the `sparseVectorIndice`? Perhaps throwing an error here would be more user-friendly?

Reply:
IsValidSparseVectorSchema() is called by isArrowDataTypeConvertible(), and isArrowDataTypeConvertible() is called by isSchemaEqual()
If IsValidSparseVectorSchema() returns false, the isArrowDataTypeConvertible() will return false and the isSchemaEqual() will throw an error "field '%s' type mis-match, milvus data type '%s', arrow data type get '%s'"

```
func isSchemaEqual(schema *schemapb.CollectionSchema, arrSchema *arrow.Schema) error {
  ........
		if !isArrowDataTypeConvertible(arrField.Type, toArrDataType, field) {
			return merr.WrapErrImportFailed(fmt.Sprintf("field '%s' type mis-match, milvus data type '%s', arrow data type get '%s'",
				field.Name, field.DataType.String(), arrField.Type.String()))
		}
    ......
}
```